libsolidity/interface/Natspec.cpp

```diff
@@ -99,16 +99,14 @@ Json::Value Natspec::devDocumentation(ContractDefinition const& _contractDef)
 		if (auto fun = dynamic_cast<FunctionDefinition const*>(&it.second->declaration()))
 		{
 			Json::Value method(devDocumentation(fun->annotation().docTags));
-			if (!method.empty())
-			{
-				// add the function, only if we have any documentation to add
-				Json::Value jsonReturn = extractReturnParameterDocs(fun->annotation().docTags, *fun);
+			// add the function, only if we have any documentation to add
+			Json::Value jsonReturn = extractReturnParameterDocs(fun->annotation().docTags, *fun);
 
-				if (!jsonReturn.empty())
-					method["returns"] = jsonReturn;
+			if (!jsonReturn.empty())
+				method["returns"] = jsonReturn;
```

Comment:
So here, if `method` is empty, this line adds a new member.

And since either it was never empty or because of the return statement it is no longer empty, the next condition will be true.

Is that how the logic should go?

Reply:
`empty()` does the following "Return true if empty array, empty object, or null; otherwise, false."

In our case method is an empty object. We want the final `if(!method.empty())` so that we don't have a function signature member when no devdoc was specified. @axic